corehq/messaging/scheduling/tests/test_recipients.py

```diff
@@ -122,16 +125,28 @@ def test_fails_with_user_data_filter_because_one_value_does_not_match(self):
                          ._passes_user_data_filter(self.mobile_user))
 
     def test_passes_with_user_case_filter(self):
-        case = create_case_2(self.domain, case_type="thing", case_json={"case_color": "red"})
-        create_case_2(self.domain, case_type=USERCASE_TYPE, external_id=self.mobile_user.user_id,
-                      case_json={"wants_email": "yes", "color": "red"}, save=True)
+        case = create_case_2(self.domain, case_type="thing", case_json={"case_color": "green"})
 
         schedule = AlertSchedule()
         schedule.use_user_case_for_filter = True
         schedule.user_data_filter = {"wants_email": ["yes"], "color": ["{case_color}"]}
         self.assertTrue(ScheduleInstance(case=case, domain=self.domain, schedule=schedule)
                         ._passes_user_data_filter(self.mobile_user))
 
+    def test_empty_string_matches_unset_property(self):
+        schedule = AlertSchedule()
+        schedule.use_user_case_for_filter = False
+        schedule.user_data_filter = {"empty": [""], "unset": ["yes", ""]}
+        self.assertTrue(ScheduleInstance(schedule=schedule)
+                        ._passes_user_data_filter(self.mobile_user))
```

Comment:
To elaborate on my comment about user data, I do think the old version of the code would pass this filter if the test set up "empty" and "unset" as part of the `CustomDataFieldsDefinition` for the domain.  But with this change, it'll work properly even if they aren't, and it solves the issue for usercase data, so it a net improvement, in my mind.